src/ekat/ekat_pack.hpp

```diff
@@ -189,13 +189,6 @@ struct Pack {
     vector_simd for (int i = 0; i < n; ++i) d[i] = v;
   }
 
-  // Init this Pack from another one.
-  template <typename T>
-  KOKKOS_FORCEINLINE_FUNCTION explicit
-  Pack (const Pack<T,n>& v) {
```

Comment:
Was this copy ctor not doing what you needed? That seems weird. IIt seems the new ctor only adds an explicit call to the scalar type ctor. But the builtin type double is implicitly constructible from float, so `d[i] = v[i]` should work when the src has a scalar type `float` and the current obj has scalar type `double`.